pkgs/by-name/ir/ironbar/package.nix

```diff
@@ -43,35 +49,44 @@ rustPlatform.buildRustPackage rec {
     hicolor-icon-theme
     gsettings-desktop-schemas
     libxkbcommon
-    openssl
-  ];
+  ] ++ (if hasFeature "http" then [ openssl ] else [ ])
+    ++ (if hasFeature "volume" then [ libpulseaudio ] else [ ])
+    ++ (if hasFeature "cairo" then [ luajit ] else [ ]);
 
-  nativeBuildInputs = [
-    pkg-config
-    wrapGAppsHook
-    gobject-introspection
-  ];
+  nativeBuildInputs = [ pkg-config wrapGAppsHook gobject-introspection ];
 
-  propagatedBuildInputs = [
-    gtk3
-  ];
+  propagatedBuildInputs = [ gtk3 ];
+
+  runtimeDeps = [ luajitPackages.lgi ];
+
+  buildNoDefaultFeatures = features != [ ];
+  buildFeatures = features;
+
+  gappsWrapperArgs = ''
+    # Thumbnailers
+    --prefix XDG_DATA_DIRS : "${gdk-pixbuf}/share"
+    --prefix XDG_DATA_DIRS : "${librsvg}/share"
+    --prefix XDG_DATA_DIRS : "${webp-pixbuf-loader}/share"
+    --prefix XDG_DATA_DIRS : "${shared-mime-info}/share"
+
+    # gtk-launch
+    --suffix PATH : "${lib.makeBinPath [ gtk3 ]}"
+  '' + (if hasFeature "cairo" then ''
+    --prefix LUA_PATH : "./?.lua;${luajitPackages.lgi}/share/lua/5.1/?.lua;${luajitPackages.lgi}/share/lua/5.1/?/init.lua;${luajit}/share/lua/5.1/\?.lua;${luajit}/share/lua/5.1/?/init.lua"
+    --prefix LUA_CPATH : "./?.so;${luajitPackages.lgi}/lib/lua/5.1/?.so;${luajit}/lib/lua/5.1/?.so;${luajit}/lib/lua/5.1/loadall.so"
+  '' else
+    "");
```

Comment:
```suggestion
  '' + lib.optionalString (hasFeature "cairo") ''
    --prefix LUA_PATH : "./?.lua;${luajitPackages.lgi}/share/lua/5.1/?.lua;${luajitPackages.lgi}/share/lua/5.1/?/init.lua;${luajit}/share/lua/5.1/\?.lua;${luajit}/share/lua/5.1/?/init.lua"
    --prefix LUA_CPATH : "./?.so;${luajitPackages.lgi}/lib/lua/5.1/?.so;${luajit}/lib/lua/5.1/?.so;${luajit}/lib/lua/5.1/loadall.so"
  '';
```

Reply:
fixed in https://github.com/NixOS/nixpkgs/pull/307726/commits/3567088dfebd8747ee10f028d1fddba07c69bf28, thanks!